model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -468,51 +468,41 @@ def __init__(
-            dims.KDim, grid=self._grid, backend=self._backend
+        self.kh_smag_e = data_alloc.zero_field(
+            self._grid, dims.EdgeDim, dims.KDim, backend=self._backend
         )
-        self.enh_smag_fac = data_alloc.allocate_zero_field(
-            dims.KDim, grid=self._grid, backend=self._backend
+        self.kh_smag_ec = data_alloc.zero_field(
+            self._grid, dims.EdgeDim, dims.KDim, backend=self._backend
         )
-        self.u_vert = data_alloc.allocate_zero_field(
-            dims.VertexDim, dims.KDim, grid=self._grid, backend=self._backend
+        self.z_nabla2_e = data_alloc.zero_field(
+            self._grid, dims.EdgeDim, dims.KDim, backend=self._backend
         )
-        self.v_vert = data_alloc.allocate_zero_field(
-            dims.VertexDim, dims.KDim, grid=self._grid, backend=self._backend
-        )
-        self.kh_smag_e = data_alloc.allocate_zero_field(
-            dims.EdgeDim, dims.KDim, grid=self._grid, backend=self._backend
-        )
-        self.kh_smag_ec = data_alloc.allocate_zero_field(
-            dims.EdgeDim, dims.KDim, grid=self._grid, backend=self._backend
-        )
-        self.z_nabla2_e = data_alloc.allocate_zero_field(
-            dims.EdgeDim, dims.KDim, grid=self._grid, backend=self._backend
-        )
-        self.z_temp = data_alloc.allocate_zero_field(
-            dims.CellDim, dims.KDim, grid=self._grid, backend=self._backend
-        )
-        self.diff_multfac_smag = data_alloc.allocate_zero_field(
-            dims.KDim, grid=self._grid, backend=self._backend
+        self.z_temp = data_alloc.zero_field(
+            self._grid, dims.CellDim, dims.KDim, backend=self._backend
         )
+        self.diff_multfac_smag = data_alloc.zero_field(self._grid, dims.KDim, backend=self._backend)
         # TODO(Magdalena): this is KHalfDim
-        self.vertical_index = data_alloc.allocate_indices(
-            dims.KDim, grid=self._grid, is_halfdim=True, backend=self._backend
+        self.vertical_index = data_alloc.index_field(
+            grid=self._grid, dim=dims.KDim, extend={dims.KDim: 1}, backend=self._backend
         )
-        self.horizontal_cell_index = data_alloc.allocate_indices(
-            dims.CellDim, grid=self._grid, backend=self._backend
+        self.horizontal_cell_index = data_alloc.index_field(
+            grid=self._grid, dim=dims.CellDim, backend=self._backend
         )
-        self.horizontal_edge_index = data_alloc.allocate_indices(
-            dims.EdgeDim, grid=self._grid, backend=self._backend
+        self.horizontal_edge_index = data_alloc.index_field(
+            grid=self._grid, dim=dims.EdgeDim, backend=self._backend
         )
-        self.w_tmp = data_alloc.allocate_zero_field(
-            dims.CellDim, dims.KDim, grid=self._grid, is_halfdim=True, backend=self._backend
+        self.w_tmp = data_alloc.zero_field(
+            self._grid, dims.CellDim, dims.KDim, extend={dims.KDim: 1}, backend=self._backend
```

Comment:
this is purely a formatting comment: I think it would be nicer if all `data_alloc` field functions have the same format when called, i.e. `zero_field` and `random_field` is called with grid and dims as  positional arguments whereas  in `index_field` they are positional